src/dataflow/DataflowAnnotatedTypeFactory.java

```diff
@@ -196,10 +198,30 @@ public Void visitMethodInvocation(MethodInvocationTree node, AnnotatedTypeMirror
                         processingEnv);
                 TypeMirror tm = type.getUnderlyingType();
                 typeNamesMap.put(tm.toString(), tm);
+                if (tm.getKind() == TypeKind.ARRAY) {
+                    replaceArrayComponentATM((AnnotatedArrayType) type);
+                }
                 type.replaceAnnotation(dataFlowType);
             }
             return super.visitMethodInvocation(node, type);
         }
+
+        /**
+         * Add the bytecode default Dataflow annotation for component type of the given {@link AnnotatedArrayType}.
+         *
+         *<p> For multi-dimensional array, this method will recursively add bytecode default Dataflow annotation to array's component type.
+         *
+         * @param arrayAtm the given {@link AnnotatedArrayType}, whose component type will be added the bytecode default.
+         */
+        private void replaceArrayComponentATM(AnnotatedArrayType arrayAtm) {
```

Comment:
Could you make this a `protected` method, so that you can share the logic with the subclass?
Then, instead of calling a static `DataflowUtils` method, call another protected method into which you extract the different behavior for the two cases. That way you should be able to reduce the duplication between the two.